### PR TITLE
Add GitHub CI workflow

### PR DESCRIPTION
**Pull request by drewnoakes**
I'm experimenting with GitHub actions for various purposes such as CI builds and automated releases. This is a first attempt.

### CODE REVIEW

**Comment by drewnoakes** [comment]
If I can get GitHub to build .NET Framework too, I think we can drop AppVeyor. The GitHub action legs are much faster than both AppVeyor and Azure Pipelines, it seems.

**Comment by drewnoakes** [comment]
FYI @iamcarbon I renamed the default branch from `master` to `main`.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  ununtu:
+    runs-on: ubuntu-latest
+    env:
+      DOTNET_NOLOGO: true
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: 7.0.x
```
Comment (iamcarbon) [inline .github/workflows/CI.yml:21]: Can we use 8.0.x here?

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  windows:
+    runs-on: windows-latest
+    env:
+      DOTNET_NOLOGO: true
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: 7.0.x
```
Comment (iamcarbon) [inline .github/workflows/CI.yml:38]: Can we use 8.0.x here?

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    - name: Restore
+      run: dotnet restore MetadataExtractor.sln
+    - name: Build 
+      run: dotnet build MetadataExtractor.sln --configuration Release
+    - name: Test net7.0
+      run: dotnet test --verbosity normal --configuration Release --no-build -f net7.0 MetadataExtractor.Tests/MetadataExtractor.Tests.csproj
```
Comment (iamcarbon) [inline .github/workflows/CI.yml:27]: Can we test against net8.0? .NET7.0 is EOL May 14, 2024

**Pull request merged**